pkg/csi/csi.go

```diff
@@ -58,19 +59,37 @@ func InitializeSidecars(ver string) error {
 		return err
 	}
 
-	sv := version.MustParseGeneric(ver)
-
-	for _, image := range si {
-		v := version.MustParseGeneric(image.Version)
-		if sv.Major() == v.Major() && sv.Minor() == v.Minor() {
-			sidecarImages = &image
-			break
+	pltVersion := version.MustParseGeneric(ver)
+
+	// this helps in not parsing the version twice in loop.
+	closestVersion := pltVersion.WithMinor(0)
+	for idx := range si {
+		siVersion := version.MustParseGeneric(si[idx].Version)
+		log.Info("verifying sidecar images version for best compatibility to platform",
+			"sidecar", siVersion, "platform", pltVersion)
+
+		// only check sidecar image versions that are not higher than platform
+		if siVersion.Major() == pltVersion.Major() && siVersion.Minor() <= pltVersion.Minor() {
+			// filter sidecar closest to platform version
+			if closestVersion.Minor() <= siVersion.Minor() {
```

Comment:
This condition is searching for the version that has the biggest minor version. From a readability perspective, it makes more sense to have written the other way around. Also, no need for the equal sign, we do not expect 2 entires to have the same version in the yaml. 

```suggestion
			if siVersion.Minor() > closestVersion.Minor  {
```

Reply:
> we do not expect 2 entires to have the same version in the yaml.
- this was for to match the initial closest version of `0`